### ISSUE TITLE
Keras with multiple outputs: cannot evaluate a metric without associated loss

### ISSUE THREAD

**Issue by durandg12** [tensorflow/tensorflow#36827]
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS 10.13.6
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.1.0-rc2-17-ge5bf8de410 2.1.0
- Python version: v3.6.7:6ec5cf24b7, Oct 20 2018, 03:02:14

**Describe the current behavior**

I am training a model which has two outputs, but I only want to train it with respect to a loss function on the first output. Nervertheless, I want to calculate a metric based on the second output. So, I pass this kind of argument to `compile`: 
`loss={'main_output': tf.keras.losses.MeanSquaredError(),}, metrics={'second_output': tf.keras.losses.MeanSquaredError()}` or `loss={'main_output': tf.keras.losses.MeanSquaredError(), 'second_output': None}, metrics={'second_output': tf.keras.losses.MeanSquaredError()}`. This does not work as intendend as the metric on `second_output`is not computed when I call the `fit` method. The first try raises a Warning which explains why:
```
WARNING:tensorflow:Output second_output missing from loss dictionary. We assume this was done on purpose. The fit and evaluate APIs will not be expecting any data to be passed to second_output.
```
I don't get why the API is done that way, for what reason no data could be passed to `second_output`just because I didn't associate a loss.

**Workaround**
I had to write a custom loss functions returning always `0.` to make things work:
```
def zero_loss(y_true, y_pred):
    return 0.
model.compile(optimizer=tf.keras.optimizers.Adam(), 
              loss={'main_output': tf.keras.losses.MeanSquaredError(), 'second_output': zero_loss}, 
              metrics={'second_output': tf.keras.losses.MeanSquaredError()})
```
But this prints a lot of useless stuff when calling `model.fit`: this prints `second_output_loss: 0.0000e+00` on each epoch.

**Describe the expected behavior**
I think that metrics on outputs without losses should be able to be computed when calling `fit`or `evaluate`.

**Code to reproduce the issue** 

```
import tensorflow as tf
import numpy as np

print('Using Tensorflow version {} (git version {})'.format(tf.version.VERSION, tf.version.GIT_VERSION))

input_data = np.random.normal(size=(1024, 5))
target_data = {'main_output' : np.random.normal(size=(1024,)), 'second_output': np.random.normal(size=(1024,))}
input_dataset = tf.data.Dataset.from_tensor_slices(input_data)
target_dataset = tf.data.Dataset.from_tensor_slices(target_data)
dataset = tf.data.Dataset.zip((input_dataset, target_dataset)).batch(16)

def make_model():
    inp = tf.keras.Input(shape=(5))
    hidden = tf.keras.layers.Dense(24, activation='relu')(inp)
    out = tf.keras.layers.Dense(1, name='main_output')(hidden)
    out2 = tf.keras.layers.Lambda(lambda x: x ** 2, name='second_output')(out)
    model = tf.keras.Model(inputs=inp, outputs=[out, out2])
    return model

epochs = 2

model = make_model()
model.compile(optimizer=tf.keras.optimizers.Adam(), 
              loss={'main_output': tf.keras.losses.MeanSquaredError(),}, 
              metrics={'second_output': tf.keras.losses.MeanSquaredError()})
model.fit(x=dataset, epochs=epochs)

model = make_model()
model.compile(optimizer=tf.keras.optimizers.Adam(), 
              loss={'main_output': tf.keras.losses.MeanSquaredError(), 'second_output': None}, 
              metrics={'second_output': tf.keras.losses.MeanSquaredError()})
model.fit(x=dataset, epochs=epochs)

model = make_model()
def zero_loss(y_true, y_pred):
    return 0.
model.compile(optimizer=tf.keras.optimizers.Adam(), 
              loss={'main_output': tf.keras.losses.MeanSquaredError(), 'second_output': zero_loss}, 
              metrics={'second_output': tf.keras.losses.MeanSquaredError()})
model.fit(x=dataset, epochs=epochs)
```

**Comment by gadagashwini** [comment]
I am able to replicate the issue with Tf 2.1.
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/40449d13fb2f790a30a5595c30ef3852/untitled399.ipynb). Thanks!

**Comment by pavithrasv** [comment]
Thank you @durandg12. This is by design today but we can look into whether the behavior can be changed. Marked this as a feature request.

**Comment by durandg12** [comment]
Thank you @pavithrasv.

To give you more context about my request, this is my real use case: the main output of my model is the element-wise multiplication of two sub-outputs. To enforce more constraints on my model, I train it with one loss on each of the two sub-outputs. Nevertheless, to evaluate the quality of the model, I still need to compute metrics on the main output, without giving it a loss.

Another detrimental effect of my actual workaround is that the use of a custom object introduces an additional burden to load the model after saving it.

**Comment by durandg12** [comment]
Any news on this?

**Comment by omsrisagar** [comment]
I am having a similar issue, any update on this?

**Comment by sushreebarsa** [comment]
@durandg12 Could you please check the updated [gist](https://colab.research.google.com/gist/sushreebarsa/0046baeebf48503e22ceb0d3f53132b3/untitled117.ipynb) in TF v2.8, Please let us know if it helps?Thanks!

**Comment by durandg12** [comment]
Hello, it seems to be working now. Thanks.

**Comment by sushreebarsa** [comment]
@durandg12 Thank you for the update!
Closing this issue as it is fixed in latest version of TensorFlow. Please feel free to reopen the issue if you still have a concern. Thank you!
